lib/rules/object-property-newline.js

```diff
@@ -61,7 +63,17 @@ module.exports = {
                         context.report({
                             node,
                             loc: firstTokenOfCurrentProperty.loc.start,
-                            message: errorMessage
+                            message: errorMessage,
+                            fix(fixer) {
+                                const comma = sourceCode.getTokenBefore(firstTokenOfCurrentProperty);
+                                const rangeAfterComma = [comma.range[1], firstTokenOfCurrentProperty.range[0]];
+
+                                if (sourceCode.text.slice(rangeAfterComma[0], rangeAfterComma[1]).trim()) {
```

Comment:
I think a comment explaining that the fixer bails when comments are present might be helpful to future contributors.